Extensions/BitmapText/JsExtension.js

```diff
@@ -0,0 +1,588 @@
+     */
+    RenderedBitmapTextInstance.getThumbnail = function(
+      project,
+      resourcesLoader,
+      object
+    ) {
+      return 'JsPlatform/Extensions/bitmapfont24.png';
+    };
+
+    /**
+     * This is called to update the PIXI object on the scene editor
+     */
+    RenderedBitmapTextInstance.prototype.update = function() {
+      const properties = this._associatedObject.getProperties();
+
+      const rawText = properties.get('text').getValue();
+      if (rawText !== this._pixiObject.text) {
+        this._pixiObject.text = rawText;
+      }
+      const opacity = properties.get('opacity').getValue();
+      this._pixiObject.alpha = opacity / 255;
+
+      const color = properties.get('color').getValue();
+      if (color !== this._pixiObject.style.fill) {
+        this._pixiObject.dirty = true;
+        this._pixiObject.style.fill = color;
+      }
+
+      const fontSize = Number(properties.get('fontSize').getValue());
+      this._pixiObject.fontSize = fontSize; // size of the text
+      this._pixiObject.style.fontSize = fontSize; //size of the bitmap font texture
+
+      const fontResourceName = properties.get('fontFamily').getValue();
+      if (this._fontResourceName !== fontResourceName) {
+        this._fontResourceName = fontResourceName;
+
+        this._pixiResourcesLoader
+          .loadFontFamily(this._project, fontResourceName)
+          .then(fontFamily => {
+            // Once the font is loaded, we can use the given fontFamily.
+            this._pixiObject.style.fontFamily = fontFamily;
+            this._pixiObject.dirty = true;
+          })
+          .catch(err => {
+            // Ignore errors
+            console.warn(
+              'Unable to load font family for RenderedBitmapTextInstance',
+              err
+            );
+          });
+      }
+
+      const wordWrap = properties.get('wordWrap').getValue() === 'true';
+      if (wordWrap !== this._pixiObject.style.wordWrap) {
+        this._pixiObject.style.wordWrap = wordWrap;
+        this._pixiObject.dirty = true;
+      }
+
+      const align = properties.get('align').getValue();
+      if (align !== this._pixiObject.align) {
+        this._pixiObject.align = align;
+        this._pixiObject.dirty = true;
+      }
+
+      this._pixiObject.position.x =
+        this._instance.getX() + this._pixiObject.width / 2;
+      this._pixiObject.position.y =
+        this._instance.getY() + this._pixiObject.height / 2;
+      this._pixiObject.rotation = RenderedInstance.toRad(
+        this._instance.getAngle()
+      );
+
+      if (this._instance.hasCustomSize() && this._pixiObject) {
+        const customWidth = this._instance.getCustomWidth();
+        if (this._pixiObject && this._pixiObject.maxWidth !== customWidth) {
+          this._pixiObject.maxWidth = customWidth;
+          this._pixiObject.dirty = true;
+        }
+      }
+      PIXI.BitmapFont.from('basicBitmap', this._pixiObject.style);
+      this._pixiObject.updateText();
```

Comment:
The canvas used by the scene in the IDE crash after some second/minutes, because the game loop take too much time.

`28[Violation] 'requestAnimationFrame' handler took <N>ms`
But without these update the bitmapText isn't on the scene.

Also the property works fine, but only once instance visible in the same time.

@4ian

Reply:
Calling `PIXI.BitmapFont.from(xxx)` is a *very expensive* operation. It involves generating an in memory canvas where the font is rendered and the a texture is created from it (see [how it's made here, it's interesting](https://pixijs.download/dev/docs/packages_text-bitmap_src_BitmapFont.js.html#line190)).

We should definitely avoid this and need to think how to create fonts only when needed.
In the case of a traditional text object, changing the font is expensive, but here in the case of BitmapText, it's expensive whenever we need to re-create the font, which can be when:
* the font is changed
* the font size is changed 
* the font color is changed
* more generally, any option impacting the `style` passed to `PIXI.BitmapFont.from`.

The second important thing is that the `BitmapFont` is holding a global map containing all the BitmapFont you created, indexed by their name. Here you create only one BitmapFont called `basicBitmap`. Instead, we should probably auto-generate a name that is containing all the font information. Something like `fontName + "-" + fontSize + "-" + fontColor + "-" + anything else` (it's a "hash" of the style in a way).
Then, when the font/font size/font color is changed, we can first check if the new font is already created (check if [`PIXI.BitmapFont.available`](https://pixijs.download/dev/docs/PIXI.BitmapFont.html#.available) contains the auto-generated name), use it if yes. And if no, we generate a new font using `PIXI.BitmapFont.from`.

The last issue will be "ok now we're generating fonts on demand, only when the style changed, but when do we release fonts that we're not using anymore?". This is more tricky, we'll probably need a method that is called on the RenderedBitmapTextInstance when it's destroyed, and store in a global object ("BitmapFontManager") the number of instances using font - so that when a font reaches a usage of `0`, we can [uninstall it](https://pixijs.download/dev/docs/PIXI.BitmapFont.html#.available). What I described is called "reference counting", you count how much instances are using your resource.

We'll probably need the same thing in the runtime object too (auto generate the font name, and track the font names usage so that we can `uninstall` the fonts that are not used anymore) - unless there is something easier to do.

Reply:
Not sure if that'll be helpful or not, but there's also some more context here about their implementation of BitmapFont, BitmapText, and the thoughts around it (along with how they're doing updates) here: https://medium.com/@bigtimebuddy/lets-talk-about-text-pixijs-244b9f95f830